week2-2/script.js

```diff
@@ -0,0 +1,275 @@
+import { HISTORY_DATA } from "./Data.js";
+
+let historyArr = [];
+let initBalance = 0;
+let plusTotal = 0;
+let minusTotal = 0;
+
+function getHistoryObj(select, text, price, status) {
+  return {
+    select: select,
+    text: text,
+    price: price,
+    status: status,
+  };
+}
+
+const historyList = document.getElementById("historylist_items");
+const template = document.getElementById("history-template");
+const plusBalance = document.getElementsByClassName("asset-type-plus")[0];
+const minusBalance = document.getElementsByClassName("asset-type-minus")[0];
+const wholeBalance = document.getElementById("balance");
+const inputCheckbox = document.getElementById("plus_p");
+const outlayCheckbox = document.getElementById("minus_p");
+
+// 초기 설정: inputCheckbox(수입)과 outlayCheckbox(지출) 모두 선택된 상태로 시작
+inputCheckbox.checked = true;
+outlayCheckbox.checked = true;
+
+inputCheckbox.addEventListener("change", loadAndDisplayData);
+outlayCheckbox.addEventListener("change", loadAndDisplayData);
+
+function loadAndDisplayData() {
+  let storedData = localStorage.getItem("historyData");
+  if (storedData) {
+    historyArr = JSON.parse(storedData); // localStorage에서 데이터를 가져와 historyArr에 할당
+  } else {
+    historyArr = JSON.parse(JSON.stringify(HISTORY_DATA)); // 새로운 배열 생성
```

Comment:
JSON.parse : 문자열 -> JSON 변환 
JSON.stringify : JSON -> 문자열 변환 
인걸로 아는데, 이걸 동시에 사용하면 기능이 상쇄되지 않을까요?? 혹시 동시에 사용한 이유가 있을까용?! 
만약 HISTORY_DATA를 historyArr로 복사하고 싶었던 거라면
```suggestion
    historyArr = [...HISTORY_DATA] 
```
이렇게 표현해줘도 좋을 것 같아요!!